src/dynamics/video_player/player/player.js

```diff
@@ -1378,37 +1381,48 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
+                        const videoParentEle = video.parentElement;
+                        this.imgEle.style.width = '100%';
+                        this.vidEle.setAttribute('crossorigin', 'anonymous')
+                        this.vidEle.muted = true;
+
+                        this.vidEle.addEventListener("loadeddata", (event) => {
+                            this.vidEle.pause();
+                            this._drawFrame(this.vidEle, 0, video.videoWidth, video.videoHeight, (canvas) => {
+                                this.imgEle.src = `${canvas.toDataURL()}`;
+                                videoParentEle.appendChild(this.imgEle);
+                            })
+                        })
+                    } catch (e) {}
+                },
+                _renderVideoFrame: function(video, isEnded) {
                     const currentPosition = this.getCurrentPosition();
                     video.style.backgroundColor = 'transparent';
                     const videoParentEle = video.parentElement;
-
-                    img = document.createElement('img');
-                    img.style.width = '100%';
-                    const imgElements = videoParentEle.querySelectorAll('img');
-
-                    const vidEle = document.createElement('video');
-                    vidEle.src = this.get("source");
-                    vidEle.setAttribute('crossorigin', 'anonymous')
-                    vidEle.muted = true;
-                    vidEle.play();
+                  
                     try {
-                        vidEle.addEventListener("loadeddata", (event) => {
-                            this._drawFrame(vidEle, currentPosition, video.videoWidth, video.videoHeight, (canvas) => {
-                                imgElements.forEach(img => {
-                                    if (img.parentNode) img.parentNode.removeChild(img)
-                                });
-                                img.src = `${canvas.toDataURL()}`;
-                                videoParentEle.appendChild(img);
-
-                                if (this.get("trackFrameTime") > currentPosition) {
-                                    video.currentTime = this.get("trackFrameTime");
-                                }
-                                this.set('trackFrameTime', currentPosition)
-                            })
-                        });
+                        this.vidEle.pause();
+                        this.imgEle.src = '';
+                        this._drawFrame(this.vidEle, currentPosition, video.videoWidth, video.videoHeight, (canvas) => {
+                            this.removeElements(videoParentEle.querySelectorAll('img'));
+                            this.imgEle.src = `${canvas.toDataURL()}`;
+                            videoParentEle.appendChild(this.imgEle);
+
+                            if (isEnded) {
+                                video.currentTime = 0;
```

Comment:
Hi @maxblessed, video refers to the actual video element being displayed in the player. I don't think we should edit `currentTime` here, changing it in the wrong place could cause unexpected side effects.